### PR TITLE
Stop deploying `cache-node-exporter` ServiceMonitor on unmanaged `Seed`s

### PR DESCRIPTION
**Pull request by rickardsjp**
**How to categorize this PR?**

/area monitoring
/kind bug

**What this PR does / why we need it**:
Gardener does not deploy the node-exporter DaemonSet on unmanaged seed clusters, i.e. seed clusters that are not simultaneously also shoot clusters.
With this change, it also stops deploying the ServiceMonitor in the `kube-system` namespace that is used by the prometheus-cache to scrape node-exporter instances on managed seeds.

**Special notes for your reviewer**:
/cc @matthias-horne @AndreasBurger 

**Release note**:

```bugfix operator
Gardener no longer deploys the `node-exporter` ServiceMonitor in the `kube-system` namespace on unmanaged `Seed`s.
```

### CODE REVIEW

File: pkg/gardenlet/controller/seed/seed/components.go [hunk 1]
```diff
@@ -585,6 +585,11 @@ func (r *Reconciler) newCachePrometheus(log logr.Logger, seed *seedpkg.Seed, see
 		return nil, fmt.Errorf("failed getting additional scrape configs: %w", err)
 	}
 
+	var serviceMonitors []*monitoringv1.ServiceMonitor
+	if seedIsShoot {
+		serviceMonitors = cacheprometheus.CentralServiceMonitors()
+	}
```
Comment (rfranzke) [inline pkg/gardenlet/controller/seed/seed/components.go]: Why are we not deploying any central service monitors if the seed is a shoot? Even if the list only contains the one for the `node-exporter` today, we probably should rather only exclude this one and not the entire list?

**Review by matthias-horne — APPROVED** [review]
Thanks for the quick fix @rickardsjp!

File: pkg/gardenlet/controller/seed/seed/components.go (hunk 1 above)
Comment (rickardsjp) [reply pkg/gardenlet/controller/seed/seed/components.go]: Good point, I moved it into the `CentralServiceMonitors` function.

File: pkg/component/observability/monitoring/prometheus/cache/servicemonitors_test.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ import (
 var _ = Describe("ServiceMonitors", func() {
 	Describe("#CentralServiceMonitors", func() {
 		It("should return the expected objects", func() {
-			Expect(cache.CentralServiceMonitors()).To(HaveExactElements(&monitoringv1.ServiceMonitor{
+			Expect(cache.CentralServiceMonitors(true)).To(HaveExactElements(&monitoringv1.ServiceMonitor{
```
Comment (rfranzke) [inline pkg/component/observability/monitoring/prometheus/cache/servicemonitors_test.go:20]: Do you want to add a unit test that ensures the list is empty for unmanaged seeds?

Comment (rickardsjp) [reply pkg/component/observability/monitoring/prometheus/cache/servicemonitors_test.go:20]: sure

**Review by rfranzke — COMMENTED** [review]
Nice

/lgtm
/approve
